gc/base/CollectorLanguageInterface.cpp

```diff
@@ -36,14 +36,26 @@ MM_CollectorLanguageInterface::writeBarrierStore(OMR_VMThread *omrThread, omrobj
 
 #if defined(OMR_GC_MODRON_CONCURRENT_MARK) || defined(OMR_GC_MODRON_SCAVENGER)
 	MM_EnvironmentStandard *env = MM_EnvironmentStandard::getEnvironment(omrThread);
-	MM_GCExtensionsBase *extensions = env->getExtensions();
+	generationalWriteBarrierStore(env, parentObject, childObject);
+	concurrentWriteBarrierStore(env, parentObject);
 #endif /* defined(OMR_GC_MODRON_CONCURRENT_MARK) | defined(OMR_GC_MODRON_SCAVENGER) */
-#if defined(OMR_GC_MODRON_CONCURRENT_MARK)
-	if (extensions->concurrentMark) {
-		extensions->cardTable->dirtyCard(env, parentObject);
-	}
-#endif /* defined(OMR_GC_MODRON_CONCURRENT_MARK) */
+}
+
+void
+MM_CollectorLanguageInterface::writeBarrierUpdate(OMR_VMThread *omrThread, omrobjectptr_t parentObject, omrobjectptr_t childObject)
+{
+#if defined(OMR_GC_MODRON_CONCURRENT_MARK) || defined(OMR_GC_MODRON_SCAVENGER)
+	MM_EnvironmentStandard *env = MM_EnvironmentStandard::getEnvironment(omrThread);
+	generationalWriteBarrierStore(env, parentObject, childObject);

```

Comment:
Can this code be broken up so that the code the scavenger and the concurrent code are not in the same #if block.  It would make the code clearer as to what was happening.